### PR TITLE
New utility library

### PR DESCRIPTION
**Pull request by Paul-Licameli**
A very tiny library of very widely used constants and templates, defined entirely in header files.

This library is no dependent on any part of wxWidgets.

### CODE REVIEW

**Review by JamesCrook — APPROVED** [review]
lgtm

File: libraries/lib-utility/MemoryX.cpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include "MemoryX.h"
+
+// Make the symbol table non-empty
+UTILITY_API void lib_utility_dummy_symbol()
```
Comment (crsib) [inline libraries/lib-utility/MemoryX.cpp:4]: Maybe we should add the concept of the header-only libraries to AudacityFunctions?

Comment (Paul-Licameli) [reply libraries/lib-utility/MemoryX.cpp:4]: I had the same thought.

However, see also my attempts to make the Mac C++17 build work without increasing the minimum SDK.  I did end up adding linked symbols to MemoryX.cpp after all.

File: libraries/lib-utility/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+did not yet provide std::make_unique.  That explains the name.
+
+Audacity now uses C++14, and yet, need arose for other anticipations of
+future standards, such as with optional.  It also provides other pervasively
+used utilities that don't correspond to things in the standard.
+
+]]#
+
+set( SOURCES
+   MemoryX.cpp
```
Comment (gera-gas) [inline libraries/lib-utility/CMakeLists.txt:19]: What have is semantic load the 'X' char in name?

Comment (Paul-Licameli) [reply libraries/lib-utility/CMakeLists.txt:19]: It was meant to suggest "Extensions"

As elsewhere, you may see X or Ex added to a few class names like wxArrayStringEx

**Pull request merged**